src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/AuditUtility.cs

```diff
@@ -298,7 +299,8 @@ private static (string severityLabel, NuGetLogCode code) GetSeverityLabelAndCode
                                 affectedGraphs.Add(graph.TargetGraphName);
                             }
 
-                            if (!auditInfo.IsDirect && resolvedDependency.Parent == currentProject)
+                            if (!auditInfo.IsDirect &&
+                                currentProject?.Data.Dependencies.Any(d => string.Equals(d.Name, packageIdentity.Id, StringComparison.OrdinalIgnoreCase)) == true)
```

Comment:
NIT: I don't think the "== true" adds legibility here.

Reply:
Since the left hand side of the `==` contains `?.`, this means the expression could be null, and `if` expressions must be bools, not null. Hence, we need to use `== true` to ensure the nullable bool is always a real bool.